src/auth/src/credentials/external_account.rs

```diff
@@ -201,14 +209,26 @@ where
 
         let audience = self.config.audience.clone();
         let subject_token_type = self.config.subject_token_type.clone();
-        let scope = self.config.scopes.clone();
+        let user_scopes = self.config.scopes.clone();
         let url = self.config.token_url.clone();
+
+        // User provides the scopes to be set on the final token they receive. The scopes they
+        // provide does not necessarily have to include the iam scope or cloud platform scope
+        // which are needed to make the call to iamcredentials endpoint. So when minting the
+        // sts token, we use the user provided or iam scope depending on whether the sts provided
+        // token is to be used with impersonation or directly.
```

Comment:
nit: you should get in the habit of capitalizing acronyms like STS or IAM. And stylizing things like `iamcredentials` with backticks. It makes things easier to read.